lib/get-esm-exports.js

```diff
@@ -1,95 +1,200 @@
+      const node = path.node
+  
+      if (node.declaration) {
+        parseDeclaration(node.declaration, exportAlias)
+      } else {
+        parseSpecifiers(node.specifiers, exportAlias, exportSpecifierNames)
+      }
+  
+      this.traverse(path)
+    },
+  
+    visitExportDefaultDeclaration(path) {
+      const node = path.node
+      const iitmRenamedExport = 'iitmRenamedExport'
+  
+      if (['ObjectExpression', 'ArrayExpression', 'Literal'].includes(node.declaration.type) && generate) {
+        const variableDeclaration = {
+          type: 'VariableDeclaration',
+          kind: 'let',
+          declarations: [
+            {
+              type: 'VariableDeclarator',
+              id: {
+                type: 'Identifier',
+                name: iitmRenamedExport,
+              },
+              init: node.declaration,
+            },
+          ],
         }
-        break
-      case 'ExportDefaultDeclaration':
-        exportedNames.add('default')
-        break
-      case 'ExportAllDeclaration':
-        if (node.exported) {
-          exportedNames.add(node.exported.name)
-        } else {
-          exportedNames.add('*')
+        path.replace(variableDeclaration)
+  
+        const newExportDefaultDeclaration = {
+          type: 'ExportDefaultDeclaration',
+          declaration: {
+            type: 'Identifier',
+            name: iitmRenamedExport,
+          },
         }
-        break
-      default:
-        warn('unrecognized export type: ' + node.type)
+        path.insertAfter(newExportDefaultDeclaration)
+      } else {
+        node.declaration.id = { type: 'Identifier', name: iitmRenamedExport }
+      }
+  
+      exportAlias['default'] = iitmRenamedExport
+      this.traverse(path)
+    },
+  
+    visitExportAllDeclaration(path) {
+      const node = path.node
+      const exportedName = node.exported ? node.exported.name : '*'
+      exportAlias[exportedName] = exportedName
+      this.traverse(path)
+    },
+  
+    visitExportSpecifier(path) {
+      const node = path.node
+      exportSpecifierNames.add(node.local.name)
+  
+      if (node.exported.name) {
+        exportAlias[node.exported.name] = node.local.name
+      } else if (node.exported.value) {
+        exportAlias[node.exported.value] = node.local.name
+      } else {
+        warn('unrecognized specifier export: ' + node.exported, FILE_NAME)
+      }
+
+      this.traverse(path)
+    },
+  })
+  
+  if (exportSpecifierNames.size !== 0 && generate) {
+    convertExportSpecifierToLet(exportSpecifierNames, ast)
+  }
+
+  if (generate) {
+    return {
+      exportAlias: exportAlias,
+      code: recast.print(ast).code
     }
   }
-  return Array.from(exportedNames)
+
+  return Object.keys(exportAlias)
 }
 
-function parseDeclaration (node, exportedNames) {
-  switch (node.declaration.type) {
+function convertExportSpecifierToLet(exportSpecifierNames, ast) {
+  recast.visit(ast, {
+    visitVariableDeclaration(path) {
```

Comment:
Visit is recursive, is it not? So wouldn't this mutate any const declarations in nested functions too? Might be better, especially for perf, to only modify top-level declarations.